### DISCUSSION TITLE
Rush Installed as part of post upgrade task is not found in the subsequent use like "commands": ["npm install -g @microsoft/rush && rush update"]

### DISCUSSION THREAD

**Discussion by gkumarcertinia** [renovatebot/renovate#30593] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

bitbucket

### Please tell us more about your question or problem

I need to run `rush update` command to update the repo-state.json for every Dependency PR created by renovate and then need to prettify & commit those repo-state.json file. So I have configured the postUpgradeTasks command like: `"postUpgradeTasks": { "commands": ["npm install -g @microsoft/rush && rush update && git add -A && rush prettier"]}` And it was working fine at version "renovateVersion":"37.432.0". 
Now these day with "renovateVersion":"38.18.12" and with same command and configuration Renovate started throwing error that `rush: not found`. 
I am using renovate/renovate:full images.
So It seem's in recent version this bug is being introduced. 
So kindly have a look and suggested the fixes.

### Logs (if relevant)

<details><summary>Logs</summary>

```

{"name":"renovate","hostname":"a6af2dd373ca","pid":9,"level":20,"logContext":"yP16BnctfmYJfZ4gKYtdK","repository":"asdf-fork","branch":"feature/codehaus-group","dep":"org.codehaus.gmavenplus:gmavenplus-plugin org.codehaus.groovy:groovy-all","err":{"cmd":"/bin/sh -c export NPM_AUTH=**redacted** NPM_REGISTRY_URL=**redacted** NPM_EMAIL= && npm install -g @microsoft/rush && rush version && rush update","stderr":"npm warn deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs\nnpm warn deprecated debuglog@1.0.1: Package no longer supported. Contact Support at https://**redacted**@2.1.2: This package is no longer supported. Please use @npmcli/package-json instead.\nnpm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported\nnpm warn deprecated osenv@0.1.5: This package is no longer supported.\nnpm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.\nnpm warn deprecated read-package-tree@5.1.6: The functionality that this package provided is now in @npmcli/arborist\nnpm warn deprecated @azure/core-http@3.0.4: deprecating as we migrated to core v2\n/bin/sh: 1: rush: not found\n","stdout":"\nadded 559 packages in 18s\n","options":{"cwd":"/tmp/renovate/repos/bitbucket-server/asdf-fork","encoding":"utf-8","env":{"HOME":"/home/ubuntu","PATH":"/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","LC_ALL":"C.UTF-8","LANG":"C.UTF-8","CONTAINERBASE_CACHE_DIR":"/tmp/renovate/cache/containerbase"},"maxBuffer":10485760,"timeout":900000},"exitCode":127,"name":"ExecError","message":"Command failed: export NPM_AUTH=**redacted** NPM_REGISTRY_URL=https://**redacted**@microsoft/rush && rush version && rush update\nnpm warn deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs\nnpm warn deprecated debuglog@1.0.1: Package no longer supported. Contact Support at https://**redacted**@2.1.2: This package is no longer supported. Please use @npmcli/package-json instead.\nnpm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported\nnpm warn deprecated osenv@0.1.5: This package is no longer supported.\nnpm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.\nnpm warn deprecated read-package-tree@5.1.6: The functionality that this package provided is now in @npmcli/arborist\nnpm warn deprecated @azure/core-http@3.0.4: deprecating as we migrated to core v2\n/bin/sh: 1: rush: not found\n","stack":"ExecError: Command failed: export NPM_AUTH=**redacted** NPM_REGISTRY_URL=https://**redacted**@microsoft/rush && rush version && rush update\nnpm warn deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs\nnpm warn deprecated debuglog@1.0.1: Package no longer supported. Contact Support at https://**redacted**@2.1.2: This package is no longer supported. Please use @npmcli/package-json instead.\nnpm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported\nnpm warn deprecated osenv@0.1.5: This package is no longer supported.\nnpm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.\nnpm warn deprecated read-package-tree@5.1.6: The functionality that this package provided is now in @npmcli/arborist\nnpm warn deprecated @azure/core-http@3.0.4: deprecating as we migrated to core v2\n/bin/sh: 1: rush: not found\n\n    at ChildProcess.<anonymous> (/usr/local/renovate/lib/util/exec/common.ts:99:11)\n    at ChildProcess.emit (node:events:531:35)\n    at ChildProcess.emit (node:domain:488:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:294:12)"},"durationMs":18612,"msg":"rawExec err","time":"2024-08-02T10:25:40.406Z","v":0}


```

</details>

**Comment by viceice** [comment]
npm install is no longer supported, use `install-npm` command

**Reply by gkumarcertinia** [reply]
Still I am getting error that `File name: undefined` if I use `install-npm` instead of `npm install`. 
Below are the detail logs:
{"name":"renovate","hostname":"c72bb9159deb","pid":9,"level":20,"logContext":"HeWbcBqqJ8SmUXeHec_Vw","repository":"asdf-fork","branch":"feature/codehaus-group","dep":"org.codehaus.gmavenplus:gmavenplus-plugin org.codehaus.groovy:groovy-all","err":{"cmd":"/bin/sh -c export NPM_AUTH=**redacted** NPM_REGISTRY_URL=**redacted** NPM_EMAIL= && install-npm @microsoft/rush && rush update && git add -A && rush prettier","stderr":"","stdout":"[11:48:18.098] INFO (94): Installing npm @microsoft/rush@5.131.0...\n[11:48:18.124] ERROR (94): ENOENT: no such file or directory, mkdir '/opt/containerbase/tools/@microsoft/rush'\n[11:48:18.125] FATAL (94): Install npm @microsoft/rush failed in 160ms.\n","options":{"cwd":"/tmp/renovate/repos/bitbucket-server/asdf-fork2","encoding":"utf-8","env":{"HOME":"/home/ubuntu","PATH":"/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","LC_ALL":"C.UTF-8","LANG":"C.UTF-8","CONTAINERBASE_CACHE_DIR":"/tmp/renovate/cache/containerbase"},"maxBuffer":10485760,"timeout":900000},"exitCode":1,"name":"ExecError","message":"Command failed: export NPM_AUTH=**redacted** NPM_REGISTRY_URL=https://**redacted**@microsoft/rush && rush update && git add -A && rush prettier\n","stack":"ExecError: Command failed: export NPM_AUTH=**redacted** NPM_REGISTRY_URL=https://**redacted**@microsoft/rush && rush update && git add -A && rush prettier\n\n    at ChildProcess.<anonymous> (/usr/local/renovate/lib/util/exec/common.ts:99:11)\n    at ChildProcess.emit (node:events:531:35)\n    at ChildProcess.emit (node:domain:488:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:294:12)"},"durationMs":521,"msg":"rawExec err","time":"2024-08-05T11:48:18.216Z","v":0}

**Reply by viceice** [reply]
oh, I need to escape the slash 🤦‍♂️

**Reply by viceice** [reply]
- https://github.com/containerbase/base/pull/3061

**Reply by viceice** [reply, marked as answer]
- https://github.com/containerbase/base/releases/tag/11.6.7
- https://github.com/renovatebot/base-image/releases/tag/7.9.0
- https://github.com/renovatebot/renovate/releases/tag/38.19.2

**Reply by viceice** [reply]
@gobardhan please try again with v38.19.2+

**Reply by gkumarcertinia** [reply]
Hi @viceice Yes it's ran successfully now after replacing `npm install -g ` with `install-npm`. Thanks for the support :)

**Comment by HonkingGoose** [comment]
We also have this issue asking for Rush.js monorepo support:

- https://github.com/renovatebot/renovate/issues/3681

That probably doesn't help your problem now though. 😉
